VideoLocker/src/main/java/org/edx/mobile/services/CourseManager.java

```diff
@@ -43,84 +31,33 @@ public static synchronized CourseManager getSharedInstance(){
-                chapter.setCategory(ochapter.category);
-            }
-            // carry this courseId with video model
-            m.setCourseId(courseId);
 
-            PathModel osequential = m.getSection();
-            if (osequential == null)
-                continue;
+    public static IBlock normalizeCourseStructure(CourseStructureV1Model courseStructureV1Model){
+        BlockModel topBlock = courseStructureV1Model.getBlockById(courseStructureV1Model.root);
+        CourseComponent course = new CourseComponent(topBlock, null);
 
-            ISequential sequential = chapter.getSequentialById(osequential.id);
-            if (sequential == null) {
-                sequential = new SequentialModel(chapter, osequential.id, osequential.name);
-                chapter.getSequential().add(sequential);
-                sequential.setCategory(osequential.category);
-
-            }
-            sequential.setSectionUrl(m.section_url);
+        for (BlockModel m : courseStructureV1Model.getDescendants(topBlock)) {
+            normalizeCourseStructure(courseStructureV1Model,m,course);
+        }
+        return course;
+    }
 
-            PathModel overtical = m.getVertical();
-            if (overtical == null)
-                continue;
+    public static void normalizeCourseStructure(CourseStructureV1Model courseStructureV1Model,
+                                                BlockModel block,
+                                                CourseComponent parent){
 
-            IVertical vertical = sequential.getVerticalById(overtical.id);
-            if (vertical == null) {
-                vertical = new VerticalModel(sequential, overtical.id, overtical.name);
-                sequential.getVerticals().add(vertical);
-                vertical.setId(overtical.id);
-                vertical.setCategory(overtical.category);
+        if ( block.isContainer() ){
+            CourseComponent child = new CourseComponent(block, parent);
+            for (BlockModel m : courseStructureV1Model.getDescendants(block)) {
+                normalizeCourseStructure(courseStructureV1Model, m, child);
             }
-            vertical.setUnitUrl(m.unit_url);
-
-            SummaryModel unit = m.getSummary();
-            if (unit == null)
-                continue;
-            unit.setVertical(vertical);
-            //FIXME - create random unit category for testing purpose.
-            unit.setCategory( unit.getId().hashCode() %2 == 0 ? "video" : "html");
-            List<String> namedPath = m.getNamed_path();
-            if (namedPath != null && !namedPath.isEmpty() && TextUtils.isEmpty(unit.getName())) {
-                unit.setName(namedPath.get(namedPath.size() - 1));
+        } else {
+            if ( BlockType.VIDEO == block.type && block.data != null
+                && (block.data  instanceof VideoData) ){
+                VideoBlockModel child = new VideoBlockModel(block, parent);
```

Comment:
What does this do? Seems like it just declares a new model object, but doesn't do anything with it?